### ISSUE TITLE
System's Text Cursor Indicator is ignored on new tab and also moving its window using only keyboard and GuiThreadInfo's GetCaretPos doesn't return its caret position

### ISSUE THREAD

**Issue by vhanla** [microsoft/terminal#6805]
# Environment

```none
Windows build number: Microsoft Windows [Versión 10.0.19041.329]
Windows Terminal version (if applicable): 1.0.1811.0

Any other software?
```

# Steps to reproduce

Enable Windows's **Text Cursor Indicator** in Control Panel.
![imagen](https://user-images.githubusercontent.com/1015823/86634592-ab1d1c00-bf97-11ea-8161-c8782b5dd8be.png)

## 1st scenario.
Open a new TAB, then you will notice that the Text Cursor Indicator is not working anymore.
![cursor issue 01b](https://user-images.githubusercontent.com/1015823/86634445-7b6e1400-bf97-11ea-8ad9-16b40ba1bae3.gif)

## 2nd scenario.
Move current Windows Terminal window using only keyboard as following:
Press `Alt`+`Space` select `Move` menu item then move it using the arrow keys. 
Now the Text Cursor Indicator stops following your typing cursor position.
![imagen](https://user-images.githubusercontent.com/1015823/86634813-f33c3e80-bf97-11ea-88bb-3d4e53236aaa.png)
![imagen](https://user-images.githubusercontent.com/1015823/86635096-4a421380-bf98-11ea-80db-a233677340fd.png)

## 3rd scenario.
Use Winapi using GetGUIThreadInfo combined with GetCaretPos, to ge curret window's caret position.
Or just use the demo project from https://www.codeproject.com/articles/34520/getting-caret-position-inside-any-application
It works even CMD and PowerShell, but not in Windows Terminal.

# Expected behavior

Keep tracking the text cursor indicator, without using workarounds to restore it as mentioned below.

And for the GetCaretPos issue, it should return the caret poition for third party tools, not only system's Text Cursor Indicator.
# Actual behavior

Text cursor indicator is ignored after those steps, and needs other steps to restore th cursor indicator tracker.

**NOTICE** To restore Text Cursor Indicator is possible by switching back and forth wih `Alt`+`Tab` or switching among tabs.

About the GetCaretPos issue, I wrote a tool to highlight the caret position for teaching purposes, specially to attract attention towards the caret position, it even follows the caret position.
![imagen](https://user-images.githubusercontent.com/1015823/86640911-55993d00-bfa0-11ea-9667-542d25f95dfe.png)
Since GetCaretPos offers that opportunity to achieve that, it is a convenient way for tools like this to help users for whatever reasons they want, like the tool in the picture above, which is working perfectly in the cmd console shell, notepad, powershell's, etc.
However, on Windows Terminal (and Visual Studio Code too) GetCaretPos doesn't work as expected.

I reported this issues since they seem related in someway.

**Comment by zadjii-msft** [comment]
These are some good reports! I thought at first that this might have been #3791, but it looks like this is actually it's own new thing. Clearly, there's something else we missed here 😅

**Comment by DHowett** [comment]
Hmmm.. yes, this should definitely be fixed.

**Comment by zadjii-msft** [comment]
Hey guess what, this was actually fixed in #12210, which should be in 1.13
